src/core/search/rax_tree.h

```diff
@@ -93,10 +93,12 @@ template <typename V> struct RaxTreeMap {
   }
 
   ~RaxTreeMap() {
-    for (auto it = begin(); it != end(); ++it) {
-      V* ptr = &(*it).second;
-      std::allocator_traits<decltype(alloc_)>::destroy(alloc_, ptr);
-      alloc_.deallocate(ptr, 1);
+    while (true) {
```

Comment:
There are two things that concern me here:

1. Looking at how `raxRecursiveFree` is implemented it calls `free_callback(raxGetData(n)` and what I introduced before seems to be on par in that respect because `V* ptr = &(*it).second;` is actually the result of `raxGetData(n)`.
2. Since we can reproduce this we should also include an isolated test to assert the root cause of the issue (my mistake not to suggest this earlier)

I would like to understand why deleting the result of `raxGetData(n)` result in segfault

Reply:
raxFree passes NULL to raxResursiveFree

Reply:
So it's not called there

Reply:
I'm still not sure what causes problem. But for sure it's not correct that we are just changing iterator data without any redis method and then passing it to the raxNext

Reply:
@BagritsevichStepan do you know how to reproduce this?

Reply:
Test was added